### PR TITLE
[SPARK-23036][SQL][TEST] Add withGlobalTempView for testing

### PR DESCRIPTION
**Pull request by xubo245**
## What changes were proposed in this pull request?

Add withGlobalTempView when create global temp view, like withTempView and withView.
And correct some improper usage.
Please see jira.
There are other similar place like that. I will fix it if community need. Please confirm it. 
## How was this patch tested?

no new test.

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/execution/GlobalTempViewSuite.scala [hunk 1]
```diff
@@ -140,8 +140,8 @@ class GlobalTempViewSuite extends QueryTest with SharedSQLContext {
 
       assert(spark.catalog.listTables(globalTempDB).collect().toSeq.map(_.name) == Seq("v1", "v2"))
     } finally {
-      spark.catalog.dropTempView("v1")
-      spark.catalog.dropGlobalTempView("v2")
+      spark.catalog.dropGlobalTempView("v1")
+      spark.catalog.dropTempView("v2")
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/GlobalTempViewSuite.scala]: Hi, @xubo245 .
Could you split the bug and new improvement into two separate PR?
Then, your PR will get reviews more easily.

**Comment by dongjoon-hyun** [comment]
BTW, please put '[SQL]' into your title.
Then, your PR will be listed under SQL category here.
- https://spark-prs.appspot.com/open-prs

**Comment by gatorsmile** [comment]
@xubo245 This is only for TEST? Then, add the tag [TEST] in your title too.

**Comment by gatorsmile** [comment]
ok to test

File: sql/core/src/test/scala/org/apache/spark/sql/execution/GlobalTempViewSuite.scala (hunk 1 above)
Comment (xubo245) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/GlobalTempViewSuite.scala]: Ok, done. Please review: https://github.com/apache/spark/pull/20250

**Comment by xubo245** [comment]
ok, done, I  put '[SQL]' into title.  @dongjoon-hyun

**Comment by xubo245** [comment]
@gatorsmile Sure. This is only for TEST. Done , I put '[SQL]' into title too

**Comment by dongjoon-hyun** [comment]
Retest this please.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/GlobalTempViewSuite.scala [hunk 2]
```diff
@@ -36,7 +36,7 @@ class GlobalTempViewSuite extends QueryTest with SharedSQLContext {
 
   test("basic semantic") {
     val expectedErrorMsg = "not found"
-    try {
+    withGlobalTempView("src") {
```
Comment (gatorsmile) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/GlobalTempViewSuite.scala]: This is wrong. Do you know why?

**Comment by gatorsmile** [comment]
LGTM

**Comment by gatorsmile** [comment]
Thanks! Merged to master/2.3

**Pull request closed without merging**